pkgs/tools/system/htop/default.nix

```
@@ -33,8 +34,7 @@ stdenv.mkDerivation rec {
   postFixup =
     let
       optionalPatch = pred: so: lib.optionalString pred "patchelf --add-needed ${so} $out/bin/htop";
-    in
-    ''
+    in lib.optionalString (!stdenv.hostPlatform.isStatic) ''
```

```suggestion
    in lib.optionalString systemdSupport ''
```

avoid the repetion on condition

---

This is not about systemd, it is about static linking. Static linking => no shared libraries => no patchelf. You may choose to build without systemd, but with lm_sensors on dynamic system, at which point readelf _should_ be called.

---

Well if you build without systemd, but with lm_sensors on dynamic system, you'd need to patchelf libsensors but not libsystemd, right? So I think you still need to make the first patchelf command conditional on `!isStatic` but the second one conditional on `systemdSupport`.

---

I think this code is correct as is. Note that there are two layers of conditionals.

Outer one, which says that `patchelf` only applies to dynamic systems.
Inner one, embedded into `optionalPatch` that says that you patch only library you built.

Honestly, I don't understand why `patchelf` is needed in first place, but that it exercise for another day. Point is that my patch is not making things worse.